### PR TITLE
Network command line switches for `dcrlnd` and `dcrlncli` which closes #70 #71

### LINKED ISSUE #70: Remove `--mainnet`: it should be the default network
This is to keep dcrlnd in line with other decred tools where by default mainnet is used unless `--testnet` or `--simnet` is specified.

### PR DESCRIPTION
**Pull request by termoose**
This is pull request for #70 and #71 as they were somewhat connected.

### CODE REVIEW

**Comment by fguisso** [comment]
Change this will impact another points of the code, I don't give an test, but just with search in files with Ack I found two files that need atention, `sample-dcrlnd.conf` and `lnd.go:144`. Please verify and test.

I don't now if we need regtest in `dcrlncli`, @matheusd can answer better.

btw, thanks for the PR, welcome to Decred!

File: cmd/dcrlncli/main.go [hunk 1]
```diff
@@ -160,11 +160,25 @@ func extractPathArgs(ctx *cli.Context) (string, string, error) {
 		return "", "", fmt.Errorf("unknown chain: %v", chain)
 	}
 
-	network := strings.ToLower(ctx.GlobalString("network"))
-	switch network {
-	case "mainnet", "testnet", "regtest", "simnet":
-	default:
-		return "", "", fmt.Errorf("unknown network: %v", network)
+	// We default to mainnet if no other options are specified
```
Comment (matheusd) [inline cmd/dcrlncli/main.go]: Please use full sentences (missing a period at the end of the sentence).

File: config.go [hunk 2]
```diff
@@ -647,21 +642,15 @@ func loadConfig() (*config, error) {
...
-			"choose one of the four"
+		str := "%s: The testnet, regtest, and simnet params" +
+			"can't be used together -- choose one of the three"
 		err := fmt.Errorf(str, funcName)
 		return nil, err
 	}
 
-	// The target network must be provided, otherwise, we won't know how to
-	// initialize the daemon.
+	// We default to mainnet if none are specified
```
Comment (matheusd) [inline config.go]: Missing final period.

File: lnd.go [hunk 3]
```diff
@@ -135,14 +135,12 @@ func Main(lisCfg ListenerCfg) error {
 	ltndLog.Infof("Version: %s, build=%s, logging=%s",
 		build.Version(), build.Deployment, build.LoggingType)
 
-	var network string
+	// We default to mainnet if none are specified
```
Comment (matheusd) [inline lnd.go]: missing final period.

**Comment by termoose** [comment]
Thanks for the great feedback! I wasn't sure what was meant by some attention needed at `lnd.go:144`? Also the `regtest` support is still there, I'm not sure if that's even usable with lightning

**Comment by fguisso** [comment]
Sorry about `lnd.go:144`, just cache in my git.
LGTM, just await the CI finish the jobs

**Comment by termoose** [comment]
I'm not able to tell why the integration test failed, but noticed while investigating that I need to also change some documentation to update them with the new arguments
- `docker/README.md`
- `docs/recovery.md`

**Comment by fguisso** [comment]
I'm working on Docker dir, just change the anothers docs and I'm update my Dockers PR

**Comment by matheusd** [comment]
Excellent! Please just squash all commits into a single one and use an appropriate title for it (such as `multi: Remove network CLI argument` or something to that effect)

**Pull request merged**